contracts/interchain-token-service/src/flow_limit.rs

```diff
@@ -0,0 +1,123 @@
+use axelar_soroban_std::ensure;
+use axelar_soroban_std::events::Event;
+use axelar_soroban_std::ttl::extend_persistent_ttl;
+use soroban_sdk::{BytesN, Env};
+
+use crate::error::ContractError;
+use crate::event::FlowLimitSetEvent;
+use crate::storage_types::DataKey;
+
+const EPOCH_TIME: u64 = 6 * 60 * 60; // 6 hours in seconds = 21600
+
+pub fn flow_limit(env: &Env, token_id: BytesN<32>) -> Option<i128> {
+    env.storage()
+        .persistent()
+        .get(&DataKey::FlowLimit(token_id))
+}
+
+pub fn set_flow_limit(
+    env: &Env,
+    token_id: BytesN<32>,
+    flow_limit: Option<i128>,
+) -> Result<(), ContractError> {
+    if let Some(limit) = flow_limit {
+        ensure!(limit > 0, ContractError::InvalidFlowLimit);
+    }
+
+    env.storage()
+        .persistent()
+        .set(&DataKey::FlowLimit(token_id.clone()), &flow_limit);
+
+    FlowLimitSetEvent {
+        token_id,
+        flow_limit,
+    }
+    .emit(env);
+
+    Ok(())
+}
+
+pub fn flow_out_amount(env: &Env, token_id: BytesN<32>) -> i128 {
+    let epoch = env.ledger().timestamp() / EPOCH_TIME;
+    env.storage()
+        .temporary()
+        .get(&DataKey::FlowOut(token_id, epoch))
+        .unwrap_or(0)
+}
+
+pub fn flow_in_amount(env: &Env, token_id: BytesN<32>) -> i128 {
+    let epoch = env.ledger().timestamp() / EPOCH_TIME;
+    env.storage()
+        .temporary()
+        .get(&DataKey::FlowIn(token_id, epoch))
+        .unwrap_or(0)
+}
+
+enum FlowDirection {
+    In,
+    Out,
+}
+
+fn add_flow(
+    env: &Env,
+    token_id: BytesN<32>,
+    flow_amount: i128,
+    direction: FlowDirection,
+) -> Result<(), ContractError> {
+    let Some(flow_limit) = flow_limit(env, token_id.clone()) else {
+        return Ok(());
+    };
+
+    let epoch = env.ledger().timestamp() / EPOCH_TIME;
+
+    let (flow_to_add_key, flow_to_compare_key) = match direction {
+        FlowDirection::In => (
+            DataKey::FlowIn(token_id.clone(), epoch),
+            DataKey::FlowOut(token_id.clone(), epoch),
+        ),
+        FlowDirection::Out => (
+            DataKey::FlowOut(token_id.clone(), epoch),
+            DataKey::FlowIn(token_id.clone(), epoch),
+        ),
+    };
+
+    let flow_to_add: i128 = env.storage().temporary().get(&flow_to_add_key).unwrap_or(0);
+    let flow_to_compare: i128 = env
+        .storage()
+        .temporary()
+        .get(&flow_to_compare_key)
+        .unwrap_or(0);
```

Comment:
Duplication of logic, just call flow_in_amount and flow_out_amount directly to get to flow_to_add and flow_to_compare

Reply:
To make it even cleaner, you can add a 
```rust
impl FlowDirection {
   fn flow(env: &Env, token_id: BytesN<32>) -> i128;

   fn reverse_flow(env: &Env, token_id: BytesN<32>) -> i128;

   fn update_flow(env: &Env, token_id: BytesN<32>, new_flow: i128);
}
```
that does the match internally instead of flow_to_add and flow_to_compare. `epoch` can be recomputed, should be cheap